### ISSUE TITLE
Install perl on NT using Mks tool kit Error

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#2160]
Migrated from [rt.perl.org#3454](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=3454) (status was 'resolved')

Searchable as RT3454$

**Comment by p5pRT** [comment]
### From pierre.cornet@veritas.com

I am trying to install perl\-5\.6\.0 on NT using MKS tool kit\.  It fails to
install\.  Could you tell how to solve the problem? Is there a way around to
install perl on NT\.  My kornshell simulator is MKS tool kit\. The error is&#8203;:
\(I see you are using the Korn shell\.  Some ksh's blow up on Configure\,
mainly on older exotic systems\.  If yours does\, try the Bourne shell
instead\.\)

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]


In \<9B545A8C3ED0D31194360008C7F4FA2F9F9B8E@&#8203;lmoxch14\.nsmg\.veritas\.com> 
Pierre Cornet wrote&#8203;:

>I am trying to install perl\-5\.6\.0 on NT using MKS tool kit\.  It fails to
>install\.  Could you tell how to solve the problem? Is there a way around to
>install perl on NT\.  My kornshell simulator is MKS tool kit\. The error is&#8203;:
>\(I see you are using the Korn shell\.  Some ksh's blow up on Configure\,
>mainly on older exotic systems\.  If yours does\, try the Bourne shell
>instead\.\)

I have successfully built perl 5\.6\.0 for NT while run the MKS ksh\,
but I did so following the README\.win32 directions\.  That is\, I 
ran vcvars32\.bat and used nmake and cl that come with the MS VC\+\+
compiler\.  Specifically I&#8203;:

&nbsp;    cd perl\-5\.6\.0
&nbsp;    cd win32
&nbsp;    vi Makefile
&nbsp;    vcvars32
&nbsp;    nmake
&nbsp;    nmake test
&nbsp;    nmake install

Note that the MKS toolkit does not come with a C compiler\, but
other POSIX on NT environments such as cygwin or mingwin do
\(they come with ports of bash and run the Configure script
while compiling everything using gcc\)\.  AFAICT running the 
Configure script under MKS ksh will not do much for you\.

BTW\, one interesting thing I noted recently was that if in
the Makefile \(or possibly the makefile\.mk if you use dmake\) is 
that if you give INST\_DRV a path\, that is something like&#8203;:

&nbsp;  INST\_DRV = d&#8203;:\\full\\windows\\path
&nbsp;  INST\_TOP = $\(INST\_DRV\)\\perl

then perl will install in d&#8203;:\\full\\windows\\path\\perl\.\.\.
In other words the INST\_DRV is not limited to just
specifying the drive letter\.  \(You can BTW use MKS sed
to do that Makefile modification for you if you'd like
though it is tricky trying to \\escape all the backslashes\)\.

HTH

Peter Prymmer
